com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/iasclient/ZigBeeIasCieExtension.java

```diff
@@ -50,10 +50,23 @@ public void extensionShutdown() {
 
     @Override
     public void nodeAdded(ZigBeeNode node) {
-        for (ZigBeeEndpoint endpoint : node.getEndpoints()) {
-            if (endpoint.getInputCluster(ZclIasZoneCluster.CLUSTER_ID) != null) {
-                endpoint.addApplication(new ZclIasZoneClient(networkManager, networkManager.getLocalIeeeAddress(), 0));
-                break;
+        if (node.isDiscovered()) {
```

Comment:
How about extracting this code into a separate method to avoid code duplication with the stuff you need to do in `nodeUpdated`?